### PR TITLE
limit amount of results produced by single scan (OSH-67)

### PR DESCRIPTION
**Pull request by lbossis**
First cut to facilitate testing

### CODE REVIEW

**Comment by kdudka** [comment]
@lbossis While `--warning-rate-limit` operates at the level of defects, `--limit-msg-len` operates at the level of events.  So they cannot be implemented in the same way.  I believe that commit 535ed2e2dcb98ee1b8df44dd9c7893a184cbb915 (pushed to your fork) makes the `--limit-msg-len` option work as intended.

I have not touched the failing tests yet.  We should pick a snippet of the excessively long messages from `nss-util-3.90.0-1.el7_9` as the testing input in order to demonstrate the typical case where the new option is useful.

The commit summary should be:
```
csgrep --limit-msg-len: limit length of event messages
```

The commit message should also include a full URL to the fixed issue:
```
Fixes: https://github.com/csutils/csdiff/issues/133
```

**Comment by lbossis** [comment]
New test csgrep("0113-gitleaks-limit-msg-len" ) output exhibit an expected behavior

**Comment by kdudka** [comment]
@lbossis you can easily sync the expected output:
```
$ cd tests/csgrep
$ ./sync.sh 0113-gitleaks-limit-msg-len
```
This will update `0113-gitleaks-limit-msg-len-stdout.txt` to reflect the code changes.  Then you can commit the changes, push to your PR and the COPR build will succeed.  Actually, what you can see in the diff is that csgrep added the ` [...]` suffix to each truncated message, and that it added a `note` event for each finding where at least one message was truncated.  These are all expected changes.

**Review by lzaoral — CHANGES_REQUESTED** [review]
@lbossis Thank you for your contribution!  Please, incorporate the changes to the commits as requested by @kdudka in https://github.com/csutils/csdiff/pull/134#issuecomment-1653115391.

The CI fails due to the following GCC warning:
```
/home/runner/work/csdiff/csdiff/src/lib/filter.cc:331:20: error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
        note.msg = std::move(noteMsg.str());
                   ^
/home/runner/work/csdiff/csdiff/src/lib/filter.cc:331:20: note: remove std::move call here
        note.msg = std::move(noteMsg.str());
                   ^~~~~~~~~~             ~
1 error generated.
```

**Comment by kdudka** [comment]
@lbossis Thanks for the update!  Now you need to squash all the fix-up commits and rewrite the commit message as described in https://github.com/csutils/csdiff/pull/134#issuecomment-1653115391. This way we end up with a clean git history where each snapshot can be built and tested.  This is crucial for using `git bisect` later on.

**Review by kdudka — APPROVED** [review]
Looks good.  Thanks for the update!

**Comment by lzaoral** [comment]
Please, ignore the `rpm-build:epel-7-ppc64le` CI failure.  That is a known Copr issue unrelated to the changes in this PR.

**Comment by lbossis** [comment]
Requested changes have been done. Someone from upstream will merge this branch to main.

File: src/lib/filter.cc [hunk 1]
```diff
@@ -296,3 +296,43 @@ void RateLimitter::flush()
...
+        ++cntTrimmed;
+    }
+
+    if (cntTrimmed) {
+        // format a message about the message trimming
+        std::ostringstream noteMsg;
+        noteMsg << "trimmed " << cntTrimmed
+            << " message(s) with length over " << maxMsgLen_;
+
+        // take location from the key even and construct a note message
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/lib/filter.cc:327]: @lbossis Could you please fix the typo as pointed out by @lzaoral?
1. fix the typo in `src/lib/filter.cc` in your local git repo
2. `git commit --amend src/lib/filter.cc`
3. `git push --force`

**Comment by kdudka** [comment]
Merging, thanks!

**Pull request merged**